src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -168,6 +175,72 @@ class Cas1SpaceBookingService(
     success(result)
   }
 
+  @Transactional
+  fun recordDepartureForBooking(
+    premisesId: UUID,
+    bookingId: UUID,
+    cas1NewDeparture: Cas1NewDeparture,
+  ): CasResult<Cas1SpaceBookingEntity> = validatedCasResult {
+    val premises = premisesService.getApprovedPremises(premisesId)
+    if (premises == null) {
+      "$.premisesId" hasValidationError "doesNotExist"
+    }
+
+    val existingCas1SpaceBooking = cas1SpaceBookingRepository.findByIdOrNull(bookingId)
+    if (existingCas1SpaceBooking == null) {
+      "$.bookingId" hasValidationError "doesNotExist"
+    }
+
+    val departureReason = departureReasonRepository.findByIdOrNull(cas1NewDeparture.reasonId)
+    if (departureReason == null) {
+      "$.cas1NewDeparture.reasonId" hasValidationError "doesNotExist"
+    } else if (departureReason.serviceScope != "approved-premises") {
+      "$.cas1NewDeparture.reasonId" hasValidationError "incorrectDepartureReasonServiceScope"
+    }
+
+    val moveOnCategory = moveOnCategoryRepository.findByIdOrNull(cas1NewDeparture.moveOnCategoryId)
+    if (moveOnCategory == null) {
+      "$.cas1NewDeparture.moveOnCategoryId" hasValidationError "doesNotExist"
+    } else if (moveOnCategory.serviceScope != "approved-premises") {
+      "$.cas1NewDeparture.moveOnCategoryId" hasValidationError "incorrectMoveOnCategoryServiceScope"
+    }
+
+    if (validationErrors.any()) {
+      return fieldValidationError
+    }
+
+    existingCas1SpaceBooking!!
+
+    if (existingCas1SpaceBooking.actualArrivalDateTime == null) {
+      return existingCas1SpaceBooking.id hasConflictError "An arrival is not recorded for this Space Booking."
+    }
+
+    if (existingCas1SpaceBooking.actualArrivalDateTime!!.isAfter(cas1NewDeparture.departureDateTime)) {
+      return existingCas1SpaceBooking.id hasConflictError "The departure date is before the arrival date."
+    }
+
+    if (existingCas1SpaceBooking.actualDepartureDateTime != null) {
+      return existingCas1SpaceBooking.id hasConflictError "A departure is already recorded for this Space Booking."
+    }
+
+    val updatedCas1SpaceBooking = existingCas1SpaceBooking.copy(
```

Comment:
is there a reason to copy the entity here instead of updating the fields directly?

I suspect it's because actualDepartureDateTime and canonicalDepartureDate are vals. We can just change these to vars (my fault when creating the initial entity)